### DISCUSSION TITLE
Using Volar internals to get codegen metadata

### DISCUSSION THREAD

**Discussion by dvh91** [vuejs/language-tools#597] [Q&A] [answered]
I'm wondering if it's theoretically possible to use Volar internal in order to parse Vue 3 syntax with typescript into a structured component metadata like prop types, slots and general comments.

It'll be especially useful considering vue-styleguidist internal parsing tools doesn't support the new syntax and Storybook also relies on these internals.

See https://github.com/vue-styleguidist/vue-styleguidist/issues/997 for further reference and the current state and plans of supporting the new syntax.

**Comment by johnsoncodehk** [comment, marked as answer]
I think TypeScript TypeChecker API can do it, I planed to explore it when implement component preview feature.

**Comment by johnsoncodehk** [comment]
It's been a long time since this request, but it's here now: https://github.com/johnsoncodehk/volar/tree/master/packages/vue-component-meta
